File: src/main/java/com/citasmedicas/appcitasmedicas/Service/AvailabilityService.java
package com.citasmedicas.appcitasmedicas.Service;


import com.citasmedicas.appcitasmedicas.dto.Response.AvailabilitySlotResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;


import java.time.LocalDate;

public interface AvailabilityService {
    Page<AvailabilitySlotResponse> getAvailableSlots(Long doctorId, LocalDate date, Long appointmentTypeId, Pageable page);
}

File: frontend/src/services/AvailabilityService.js
// ========== AVAILABILITY ==========
export const availabilityService = {
  getAvailableSlots: (doctorId, date, appointmentTypeId, page = 0, size = 20) => 
    api.get(`/availability/doctors/${doctorId}?date=${date}&appointmentTypeId=${appointmentTypeId}&page=${page}&size=${size}`)
};